rabit/rabit/Goal/GoalAdd/GoalAddViewModel.swift

```diff
@@ -55,5 +64,42 @@ private extension GoalAddViewModel {
             .map { $0.0.selectedTime }
             .bind(to: navigation.showTimeSelectView)
             .disposed(by: disposeBag)
+        
+        let goal = Observable
+                    .combineLatest(
+                        goalTitleInput,
+                        goalSubtitleInput,
+                        selectedPeriod,
+                        selectedTime
+                    )
+                    .withUnretained(self)
+                    .map {
+                        Goal(
+                            title: $1.0,
+                            subtitle: $1.1,
+                            progress: .zero,
+                            period: $1.2,
+                            certTime: $1.3,
+                            category: $0.category.title
+                        )
+                    }
+                    .share()
```

Comment:
```suggestion
        let goal = Observable
            .combineLatest(
                  goalTitleInput, 
                  goalSubtitleInput, 
                  selectedPeriod, 
                  selectedTime
             )
            .withUnretained(self) { ($1.0, $1.1, .zero, $1.2, $1.3, $0.category.title) }
            .map(Goal.init(title:subtitle:progress:certTime:target:category:))
            .share()
```

이거 말씀하신거 같네요! 그런데 생성자의 `progress: .zero`는 어떤 의미인가요? progress는 Int 타입인데 .zero가 가능한건지 궁금합니다!

그리고 share()를 쓰든 안쓰든 `goal`는 Observable<Goal> 타입인거 같은데, 혹시 어떤 이유로 붙이는건지 궁금합니다!ㅎㅎ 이건 개인적인 질문입니다..ㅎㅎ

Reply:
suggested change 부분에 `.zero`는 해당 값이 어떤 의미를 가지는지 몰라서 일단 대충 작성한거라 `commit suggestion` 바로 하시면 안될거에요 아마도!

Reply:
`.zero` 는 맨 처음 생성할 때 progress는 값이 없는 상태여야 하기 때문에 사용했어요! target = 분모, progress = 분자 이렇게 보면 분자의 초기값을 0으로 둔 셈으로 보시면 됩니다!

Reply:
`share()` 같은 경우는 위에서 Goal 타입 인스턴스를 생성하는 Observable 시퀀스는 구독할 때마다 Observable을 생성하는 형태인데, 이렇게 되면 여러 곳에서 해당 시퀀스를 구독할 때마다 불필요하게 Observable이 너무 많이 생성되기 때문에 이걸 방지하려고 사용했어요!
![image](https://user-images.githubusercontent.com/68586291/194279645-6fc19c0a-24ee-4cbc-98af-858acdbd0d46.png)

`share()` 을 사용하면 한 번 생성된 시퀀스를 여러 곳에서 공유할 수 있는 형태이기 때문에, 이후의 여러 구독 객체들은 해당 시퀀스를 공유하게 됩니다! 
(사실 여기서는 크게 의미없는 사용이긴 한데... 저도 RxSwift 학습하다가 저렇게 `share()` 을 사용한걸 보고 궁금해서 이번에 사용해봤어요 ㅋㅋ)
![image](https://user-images.githubusercontent.com/68586291/194279699-7850579c-5c6e-4280-af59-e411f0dd60dc.png)

Reply:
@Hansolkkim 
말나온김에 ㅋㅋ
이전까지는 잘 몰라서.. RxSwift 오퍼레이터 사용한 코드는 흐름만 이래하는 식이었는데, `share()` 사용 보고 Observable 동작원리 한 번 더 공부하다보니까 이런 식으로 불필요하게 시퀀스가 생성되는 흐름이 있었겠구나(?)하는 생각도 들었어요 ㅋㅋ 필요한 기능 우선구현 끝나면 이 부분 한번 더 쭉 점검해도 좋을 것 같아요!

Reply:
오 자세한 설명 감사합니다.. 저도 이렇게 사용하려고 만드는 Onservable들이 많은데 share()로 처리하는게 더 맞겠네요!!